### DISCUSSION TITLE
Does the number of properties affect performance ?

### DISCUSSION THREAD

**Discussion by PhungQuocLuong-git** [vesoft-inc/nebula#5342] [Q&A] [answered]
Hello,
I wonder if storing too many properties in 1 vertex is good or bad for graph performance?

For example, if I have a PRODUCT vertex that has 80 properties, should I separate some properties of that PRODUCT vertex and save it in DETAILS vertex to reduce the number of properties in PRODUCT vertex? 
Is there another method to solve this problem?

Thanks :))

**Comment by wey-gu** [comment, marked as answer]
Dear @PhungQuocLuong-git ,

Generally, we should be careful when deciding to put info as properties, and yes more properties will require more resources. That is to fetch more data when returning properties/to cache less vertices/edges in same size of block caches of the rocksdb.

When properties are needed, don't worry, just put them there, it should work but just be more costly(spawning the cluster with more ram will help). One thing to be highlighted is we should always precisely express your query to only needed properties(when none of the prop is needed, don't return).

Regarding whether to put certain things as property or a vertex/edge, it depends on the visit pattern of the system. You could try both and profile their cost with `PROFILE` as the prefix of the queries.

BR//Wey
